cl/citations/tasks.py

```diff
@@ -114,42 +119,75 @@ def find_citations_and_parantheticals_for_recap_documents(
 def find_citations_and_parentheticals_for_opinion_by_pks(
     self,
     opinion_pks: List[int],
+    disconnect_pg_signals: bool = False,
 ) -> None:
     """Find citations and authored parentheticals for search.Opinion objects.
 
     :param opinion_pks: An iterable of search.Opinion PKs
+    :param disconnect_pg_signals: True if ParentheticalGroup post_save and
+        post_delete signals should be disconnected; useful in batch jobs
+        from the `find_citations` command
+
     :return: None
     """
     opinions: QuerySet[Opinion, Opinion] = Opinion.objects.filter(
         pk__in=opinion_pks
     )
-    for opinion in opinions:
+    # delivery_info does not exist in test environment
+    children_queue = (self.request.delivery_info or {}).get(
+        "routing_key", settings.CELERY_ETL_TASK_QUEUE
+    )
+
+    for index, opinion in enumerate(opinions):
+        if disconnect_pg_signals:
+            disconnect_parenthetical_group_signals()
+
         try:
-            # delivery_info does not exist in test environment
-            children_queue = (self.request.delivery_info or {}).get(
-                "routing_key", settings.CELERY_ETL_TASK_QUEUE
-            )
             store_opinion_citations_and_update_parentheticals(
                 opinion,
                 children_queue,
             )
         except ResponseNotReady as e:
             # Threading problem in httplib.
             raise self.retry(exc=e, countdown=2)
+        except OperationalError:
+            # delay deadlocked tasks, and continue regular process
+            store_opinion_citations_and_update_parentheticals.apply_async(
+                (opinion.id, children_queue), countdown=60
+            )
+        except Exception as e:
+            # do not retry the whole loop on an unknown exception
+            end_index = max(len(opinions) - 1, index + 1)
+            ids = [o.id for o in opinions[end_index:]]
+            if ids:
+                raise self.retry(
+                    exc=e, countdown=60, kwargs={"opinion_pks": ids}
+                )
+        finally:
+            if disconnect_pg_signals:
+                reconnect_parenthetical_group_signals()
 
 
+@app.task(bind=False, max_retries=5, ignore_result=True)
 def store_opinion_citations_and_update_parentheticals(
-    opinion: Opinion,
+    opinion_or_pk: Opinion | int,
```

Comment:
Actually, we should really avoid passing opinions for precisely this reason. Let's make a new task that accepts only the ID, and update the code to use it. Then in a few hours, let's remove this task.